### PR TITLE
feat(platform/azure): Recreate Auto-Approve on PR-Update

### PR DESCRIPTION
**Pull request by horihel**
## Changes

This change will allow renovate to re-approve a PR on azure if autoApprove is enabled. This should solve #15664

This makes autoApprove behavior more intuitive, as updated PRs now have the same constraints as fresh PRs.

## Context

See description and discussion in the issue #15664 .

Code (and test) is mostly copied/modified from the corresponding snippet in createPR(). 

As these are my very first lines of typescript, the code (and especially the test) might be lacking in quality. If that is the case, I'm thankful for pointers.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [X] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/azure/index.spec.ts [hunk 1]
```diff
@@ -881,6 +881,45 @@ describe('modules/platform/azure/index', () => {
...
+      };
+      const prUpdateResult = {
+        reviewerUrl: prResult.createdBy.url,
+        vote: AzurePrVote.Approved,
+        isFlagged: false,
+        isRequired: false,
+      };
+      const updateFn = jest
+        .fn(() => prUpdateResult)
+        .mockName('createPullRequestReviewer');
```
Comment (viceice) [inline lib/modules/platform/azure/index.spec.ts:902]: remove mock name, not needed

File: lib/modules/platform/azure/index.ts [hunk 2]
```diff
@@ -527,6 +528,25 @@ export async function updatePr({
   } else if (state === 'closed') {
     objToUpdate.status = PullRequestStatus.Abandoned;
   }
+  if (platformOptions?.autoApprove) {
+    const pr: GitPullRequest = await azureApiGit.getPullRequest(
+      config.repoId,
+      prNo,
+      config.project
+    );
```
Comment (viceice) [inline lib/modules/platform/azure/index.ts:536]:
```suggestion
    const pr = await azureApiGit.getPullRequestById(pullRequestId, config.project);
```

Comment (horihel) [reply lib/modules/platform/azure/index.ts:536]: I'm confused... pullRequestId doesn't exist here. Was your intention just to remove repoId?

Comment (horihel) [reply lib/modules/platform/azure/index.ts]: i replaced pullRequestId with prNo which I think is what you wanted to suggest

File: lib/modules/platform/azure/index.spec.ts [hunk 3]
```diff
@@ -881,6 +881,43 @@ describe('modules/platform/azure/index', () => {
...
+        isFlagged: false,
+        isRequired: false,
+      };
+      const updateFn = jest.fn(() => prUpdateResult);
+      azureApi.gitApi.mockImplementationOnce(
+        () =>
+          ({
+            updatePullRequest: jest.fn(() => prResult),
+            createPullRequestReviewer: updateFn,
+            getPullRequest: jest.fn(() => ({
```
Comment (JamieMagee) [inline lib/modules/platform/azure/index.spec.ts:906]:
```suggestion
            getPullRequestById: jest.fn(() => ({
```

File: lib/modules/platform/azure/index.ts (hunk 2 above)
Comment (viceice) [reply lib/modules/platform/azure/index.ts]: yes, sorry, forgot to rename the variable after copy 🙃

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**